src/index.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
+
+const changedArray = (a: Array<unknown> = [], b: Array<unknown> = []) =>
+  a.length !== b.length || a.some((item, index) => !Object.is(item, b[index]))
+
+interface FallbackProps {
+  error: Error
+  resetErrorBoundary: (...args: Array<unknown>) => void
+}
+
+interface ErrorBoundaryPropsWithComponent {
+  onResetKeysChange?: (
+    prevResetKeys: Array<unknown>,
+    resetKeys: Array<unknown>,
+  ) => void
+  onReset?: (...args: Array<unknown>) => void
+  onError?: (error: Error, info: {componentStack: string}) => void
+  resetKeys?: Array<unknown>
+  FallbackComponent: React.ComponentType<FallbackProps>
+}
+
+declare function FallbackRender(
+  props: FallbackProps,
+): React.ReactElement<
+  unknown,
+  string | React.FunctionComponent | typeof React.Component
+> | null
+
+interface ErrorBoundaryPropsWithRender {
+  onResetKeysChange?: (
+    prevResetKeys: Array<unknown>,
+    resetKeys: Array<unknown>,
+  ) => void
+  onReset?: (...args: Array<unknown>) => void
+  onError?: (error: Error, info: {componentStack: string}) => void
+  resetKeys?: Array<unknown>
+  fallbackRender: typeof FallbackRender
+}
+
+interface ErrorBoundaryPropsWithFallback {
+  onResetKeysChange?: (
+    prevResetKeys: Array<unknown>,
+    resetKeys: Array<unknown>,
+  ) => void
+  onReset?: (...args: Array<unknown>) => void
+  onError?: (error: Error, info: {componentStack: string}) => void
+  resetKeys?: Array<unknown>
+  fallback: React.ReactElement<
+    unknown,
+    string | React.FunctionComponent | typeof React.Component
+  > | null
+}
+
+type ErrorBoundaryProps =
+  | ErrorBoundaryPropsWithFallback
+  | ErrorBoundaryPropsWithComponent
+  | ErrorBoundaryPropsWithRender
+
+type ErrorBoundaryState = {error: Error | null}
+const initialState = {error: null}
```

Comment:
👋 you need to specify:
```typescript
const initialState: ErrorBoundaryState = { error: null }
```
Otherwise the typeof `state = initialState` will be `{ error: null }`. This is what the rule `@typescript-eslint/no-unnecessary-condition` is complaining about, `error !== null` is unnecessary because it's typed as `error: null` so from TS perspective `error !== null` will never be true.

This is because of the generic `{error: Error | null}`. Since `{ error: null }` extends `ErrorBoundaryState` then it's inferred as `{ error: null }` when you do `state = initialState`.

Alternatively, you could set in line 70:
```typescript
state: ErrorBoundaryState = initialState
```
And then you'll be able to remove the comments regarding `@typescript-eslint/no-unnecessary-condition`

I hope this makes sense 😅

Reply:
Makes perfect sense. And it works. Thank you!